samples/openthread/coap_client/boards/nrf5340dk_nrf5340_cpuapp.overlay

```diff
@@ -0,0 +1,53 @@
+/* Copyright (c) 2021 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+&adc {
+	status = "disabled";
```

Comment:
Why those peripherals are enabled in the first place? It seems super strange to disable tens of peripherals for each example.

Reply:
These are enabled by default, aren't they?
It is a [common practice](https://github.com/nrfconnect/sdk-nrf/blob/master/samples/openthread/coap_client/boards/nrf52840dk_nrf52840.overlay) to disable unused peripherals for coap_client sample.

Reply:
Everything is enabled by default to limit actions needed to activate a chosen feature. The case for coap_client sample, especially MTD variant, is to present the minimum current consamption. These is why we are disabling unused peripheral by this overlay for this sample.

Reply:
I just questioned scheme of enabling peripherals by default (and if I'm not mistaken, only selected peripherals) - not the necessity of this PR.

Is this really the case in Zephyr for all MCUs? I would expect applications to enable peripherals only if they are needed, not in another way around.